vars/infra.groovy

```diff
@@ -1,5 +1,13 @@
 #!/usr/bin/env groovy
 
+Boolean isRunsOnJenkinsAzure() {
+    return isJenkinsIO() || isTrusted()
+}
+
+Boolean isJenkinsIO() {
```

Comment:
I don't think this one is useful. It should be rare enough that `runningOnJenkinsInfra && !trusted` works for this purpose just as well, and we don't introduce an ambiguity with the name: note how trusted.ci also is a `jenkins.io` domain.